src/components/ViewPackagePage/components/markdown-viewer.tsx

```diff
@@ -1,15 +1,39 @@
-import React, { useEffect } from "react"
+import React, { useEffect, useState } from "react"
 import Markdown from "react-markdown"
 import remarkGfm from "remark-gfm"
 
 import { useShikiHighlighter } from "@/hooks/use-shiki-highlighter"
+import { Check, Clipboard, Copy, CopyPlus } from "lucide-react"
+import { useToast } from "@/hooks/use-toast"
 
 export default function MarkdownViewer({
   markdownContent,
 }: {
   markdownContent: string
 }) {
   const { highlighter } = useShikiHighlighter()
+  const [copiedCode, setCopiedCode] = useState<string | null>(null)
+  const { toast } = useToast()
+
+  const handleCopy = async (text: string) => {
+    try {
+      await navigator.clipboard.writeText(text)
+      setCopiedCode(text)
+      toast({
+        title: "Copied to clipboard",
+        description: "Your code has been copied successfully.",
+      })
```

Comment:
Even without the toast it is good enough I think. Cause the checked symbol gives the user info that the text is copied. We don't want another toast to convey the message.